bloom.c

```diff
@@ -119,83 +121,116 @@ static void add_key_to_filter(struct bloom_key *key,
 	}
 }
 
+static void fill_filter_from_graph(struct commit_graph *g,
+				   struct bloom_filter *filter,
+				   struct commit *c)
+{
+	uint32_t lex_pos, prev_index, next_index;
+
+	while (c->graph_pos < g->num_commits_in_base)
+		g = g->base_graph;
+
+	lex_pos = c->graph_pos - g->num_commits_in_base;
+
+	next_index = get_be32(g->chunk_bloom_indexes + 4 * lex_pos);
+	if (lex_pos)
+		prev_index = get_be32(g->chunk_bloom_indexes + 4 * (lex_pos - 1));
+	else
+		prev_index = 0;
+
+	filter->len = next_index - prev_index;
+	filter->data = (uint64_t *)(g->chunk_bloom_data + 8 * prev_index + 12);
```

Comment:
Mark this as a bug/todo for later.  
This is going to cause problems for platforms (CPUs) that don't allow non-aligned access.
If we assume that `g->chunk_bloom_data` is an allocated pointer, malloc will give us a quad-word aligned pointer.  Then adding a multiple of 8 for the prev_index, still keeps us quad-aligned.  But the `+12` won't be.